daal4py/sklearn/utils/validation.py

```diff
@@ -62,3 +69,431 @@ def _daal_assert_all_finite(X, allow_nan=False, msg_dtype=None):
+
+    order : 'F', 'C' or None (default=None)
+        Whether an array will be forced to be fortran or c-style.
+        When order is None (default), then if copy=False, nothing is ensured
+        about the memory layout of the output array; otherwise (copy=True)
+        the memory layout of the returned array is kept as close as possible
+        to the original array.
+
+    copy : boolean (default=False)
+        Whether a forced copy will be triggered. If copy=False, a copy might
+        be triggered by a conversion.
+
+    force_all_finite : boolean or 'allow-nan', (default=True)
+        Whether to raise an error on np.inf, np.nan, pd.NA in array. The
+        possibilities are:
+
+        - True: Force all values of array to be finite.
+        - False: accepts np.inf, np.nan, pd.NA in array.
+        - 'allow-nan': accepts only np.nan and pd.NA values in array. Values
+          cannot be infinite.
+
+        .. versionadded:: 0.20
+           ``force_all_finite`` accepts the string ``'allow-nan'``.
+
+        .. versionchanged:: 0.23
+           Accepts `pd.NA` and converts it into `np.nan`
+
+    ensure_2d : boolean (default=True)
+        Whether to raise a value error if array is not 2D.
+
+    allow_nd : boolean (default=False)
+        Whether to allow array.ndim > 2.
+
+    ensure_min_samples : int (default=1)
+        Make sure that the array has a minimum number of samples in its first
+        axis (rows for a 2D array). Setting to 0 disables this check.
+
+    ensure_min_features : int (default=1)
+        Make sure that the 2D array has some minimum number of features
+        (columns). The default value of 1 rejects empty datasets.
+        This check is only enforced when the input data has effectively 2
+        dimensions or is originally 1D and ``ensure_2d`` is True. Setting to 0
+        disables this check.
+
+    estimator : str or estimator instance (default=None)
+        If passed, include the name of the estimator in warning messages.
+
+    Returns
+    -------
+    array_converted : object
+        The converted and validated array.
+    """
+    if force_all_finite not in (True, False, 'allow-nan'):
+        raise ValueError('force_all_finite should be a bool or "allow-nan"'
+                         '. Got {!r} instead'.format(force_all_finite))
+
+    if estimator is not None:
+        if isinstance(estimator, str):
+            estimator_name = estimator
+        else:
+            estimator_name = estimator.__class__.__name__
+    else:
+        estimator_name = "Estimator"
+    context = " by %s" % estimator_name if estimator is not None else ""
+
+    array_orig = array
+
+    try:
+        from pandas import DataFrame, option_context
+        if isinstance(array, DataFrame) and force_all_finite != 'allow-nan':
+            if force_all_finite:
+                with option_context('mode.use_inf_as_na', True):
+                    if array.isna().any().any():
```

Comment:
We have such functionality in DAAL. Can we use this for check? @Alexsandruss What do you think?

Reply:
Now in ```_daal_assert_all_finite(X)``` any ```X``` is converted to ```numpy.array``` and it slows down everything for pandas case. So, we need to change it to use DAAL functionality

Reply:
Yes, it looks like you need to change. Because the implementation in DAAL can work with heterogeneous data? Am I right, @Alexsandruss ?

Reply:
Yes, _daal_assert_all_finite should handle pandas w/o conversions like _daal_train_test_split, which is WIP.